File: src/main/kotlin/dev/jefy/connectpro/user/applicaion/query/UserQueryImpl.java
package dev.jefy.connectpro.user.applicaion.query;

import org.springframework.security.core.Authentication;
import org.springframework.security.core.context.SecurityContextHolder;
import org.springframework.stereotype.Service;

import dev.jefy.connectpro.user.applicaion.dtos.UserResponse;
import dev.jefy.connectpro.user.domain.model.AuthUser;
import dev.jefy.connectpro.user.domain.repository.UserRepository;
import lombok.RequiredArgsConstructor;

/**
 * @author Jôph Yamba
 */
@Service
@RequiredArgsConstructor
public class UserQueryImpl implements UserQuery {
    private final UserRepository userRepository;

    @Override
    public UserResponse getAuthenticatedUser() {
        Authentication auth = SecurityContextHolder.getContext().getAuthentication();
        if (auth != null && auth.isAuthenticated() && auth.getPrincipal() instanceof AuthUser user) {
            return UserResponse.fromUser(user);
        }
        return null;
    }
}

File: src/main/kotlin/dev/jefy/connectpro/user/application/query/UserQueryImpl.kt
package dev.jefy.connectpro.user.application.query

import dev.jefy.connectpro.user.application.dtos.UserResponse
import dev.jefy.connectpro.user.application.dtos.toUserResponse
import dev.jefy.connectpro.user.application.exceptions.UserNotFoundException
import dev.jefy.connectpro.user.domain.model.AuthUser
import org.springframework.security.core.Authentication
import org.springframework.security.core.context.SecurityContextHolder
import org.springframework.stereotype.Service
import org.springframework.transaction.annotation.Transactional

@Service
@Transactional(readOnly = true)
class UserQueryImpl : UserQuery {

    override fun getAuthenticatedUser(): UserResponse {
        val auth: Authentication? = SecurityContextHolder.getContext().authentication
        
        if (auth != null && auth.isAuthenticated && auth.principal is AuthUser) {
            val user = auth.principal as AuthUser
            return user.toUserResponse()
        }
        throw UserNotFoundException()
    }
}